fdbserver/worker.actor.cpp

```diff
@@ -1051,12 +1055,27 @@ ACTOR Future<Void> workerServer(
 		wait(waitForAll(recoveries));
 		recoveredDiskFiles.send(Void());
 
-		errorForwarders.add( registrationClient( ccInterface, interf, asyncPriorityInfo, initialClass, ddInterf, rkInterf, degraded, errors, locality, dbInfo ) );
+		errorForwarders.add( registrationClient( ccInterface, interf, asyncPriorityInfo, initialClass, ddInterf, rkInterf, degraded, errors, locality, dbInfo, connFile, issues) );
 
 		TraceEvent("RecoveriesComplete", interf.id());
 
 		loop choose {
-
+			when( UpdateServerDBInfoRequest req = waitNext( interf.updateServerDBInfo.getFuture() ) ) {
+				Optional<Endpoint> notUpdated;
+				if(!ccInterface->get().present() || req.dbInfo.clusterInterface != ccInterface->get().get() || (req.dbInfo.infoGeneration < dbInfo->get().infoGeneration && dbInfo->get().clusterInterface == ccInterface->get().get())) {
+					notUpdated = interf.updateServerDBInfo.getEndpoint();
+				}
+				if(ccInterface->get().present() && req.dbInfo.clusterInterface == ccInterface->get().get() && (req.dbInfo.infoGeneration > dbInfo->get().infoGeneration || dbInfo->get().clusterInterface != ccInterface->get().get())) {
+					ServerDBInfo localInfo = req.dbInfo;
+					TraceEvent("GotServerDBInfoChange").detail("ChangeID", localInfo.id).detail("MasterID", localInfo.master.id())
+					.detail("RatekeeperID", localInfo.ratekeeper.present() ? localInfo.ratekeeper.get().id() : UID())
+					.detail("DataDistributorID", localInfo.distributor.present() ? localInfo.distributor.get().id() : UID());
+
+					localInfo.myLocality = locality;
+					dbInfo->set(localInfo);
+				}
+				errorForwarders.add(success(broadcastDBInfoRequest(req, SERVER_KNOBS->DBINFO_SEND_AMOUNT, notUpdated, true)));
```

Comment:
In case `req.dbInfo.infoGeneration < dbInfo->get().infoGeneration`, is it still necessary to forward the message?

Reply:
good suggestion, you are right it is not necessary